### DISCUSSION TITLE
Plans for web & WASM?

### DISCUSSION THREAD

**Discussion by erlend-sh** [buzz-language/buzz#118] [Ideas]
Given your history with Fengari, I wonder if you have plans for WebAssembly with buzz.

We are making an open source game with a WASM-based scripting system:
https://github.com/fishfolk/jumpy/discussions/489

Buzz could conceivably be used for scripting in our game, if it supported WASM exports.

**Comment by giann** [comment]
I’m not sure Buzz is yet a good fit for gamedev. Its VM is relatively slow depending on what bench you run.

I guess you could use it for non-performance critical scripting. In theory you could already build it for WASM thanks to zig. But i’ve not tried it yet.

Alternatively I’m exploring the idea of a AOT compiler with LLVM (#115). Such a compiler would easily target WASM and be much faster.

**Reply by erlend-sh** [reply]
Thanks. Looks like you’ve moved on to a JIT approach?

Meanwhile, [Cyber](https://github.com/fubark/cyber) just popped up as a (Zig-based) language specifically intended for gamedev & WASM as a core use case.

**Reply by giann** [reply]
Yes I'm working on the JIT compiler using LLVM. Even though it's in master it's very much a work in progress.

Cyber looks great yes 👍
